src/main/kotlin/org/eln2/libelectric/sim/electrical/generic/GenericCircuit.kt

```diff
@@ -0,0 +1,54 @@
+package org.eln2.libelectric.sim.electrical.generic
+
+import org.eln2.libelectric.sim.electrical.mna.component.Component
+
+interface GenericCircuit {
+    fun addComponent(c: GenericComponent)
+    fun removeComponent(c: GenericComponent)
+    fun simulate(timeDelta: Double)
+    fun connectComponents(
+        localComponent: GenericComponent,
+        localConnection: GenericConnection,
+        remoteComponent: GenericComponent,
+        remoteConnection: GenericConnection
+    )
+    fun disconnectComponents(localComponent: GenericComponent, remoteComponent: GenericComponent)
+    val backingImplementation: String
+}
+
+open class GenericComponent {
+    // This feels like a subpar solution as opposed to having some way for the system to call back into the backing class.
+    fun getVoltageAcross(point1: GenericConnection, point2: GenericConnection): Double {
+        return when (point1) {
+            is Component -> {
+                return point1.node(0).potential - point2.node(1).potential
+            } else -> {
+                0.0
+            }
+        }
+    }
+    fun getCurrentAcross(point1: GenericConnection, point2: GenericConnection): Double {
```

Comment:
This isn't usually a sensible calculation; KCL demands that the current _at_ a node is zero--it makes more sense to talk about the current in "loops" (ears of the graph). It makes more sense for sources--for which one output of a VS is its current, and is, of course, the _input_ for a CS.

For some components, this doesn't even make sense--e.g., across a transformer, where the current is 0.

More commonly, pairs of terminals often satisfy the "port condition": current into one is equal to the current going out of the other. Asking for the current _through_ a port is sensible, and all of our current components are two-terminal ports. (The transformer example would have two independent ports.) Some components aren't ports--notably transistors--but nonetheless satisfy the KCL (the sum of all currents is 0).

For more discourse on this, see [our current `Port` class](https://github.com/age-series/libelectric/blob/ed9d172a096fe0971cd86deb800d659931c98915/src/main/kotlin/org/eln2/libelectric/sim/electrical/mna/component/Component.kt#L207-L244).

Reply:
Redoubling on it: _this_ little issue is the reason I'm hesitant to make this be the standard. The rest are ancillary :)